### ISSUE TITLE
Docs theme with additional Plugins

### ISSUE THREAD

**Issue by derHodrig** [nuxt/content#746]
Hello,
i want to implement a vue-mermaid plugin inside my nuxt-docs app. 
Since the nuxt.config.js got exported I cant import plugins like

```js
import theme from '@nuxt/content-theme-docs'

export default theme({
  plugins: [
        // https://yarnpkg.com/package/vue-mermaid#readme
        '~/plugins/vue-mermaid.js'
  ],
})
```

The App tries then to open a plugin folder relative to the node_modules nuxt-docs-theme package...

> Plugin not found: myApp\node_modules\@nuxt\content-theme-docs\plugins\vue-mermaid.js

How to implement Plugins with this new architecture?

**Comment by derHodrig** [comment]
With that approach i got a render error

_Unexpected Token 'export'_

Which must be the "export" from nuxt.config.js. I tried to provide a codesandbox for you but it seems there is an big issue with performance (It kills my browser)... If you could run the sandbox, then there are 2 points of interest. nuxt.config.js and the index.md in content/de, where the vue-mermaid component gets called.

https://codesandbox.io/s/silly-resonance-9x69z


I know this is not a solution but a workaround. I could try to extract the docs theme into a standalone nuxt app, but idk if this is okay for you guys, since i also have to remove the brandings.

**Comment by derHodrig** [comment]
I tried that package and the linked remark-graphviz, both arent working. Regarding the issueboard i will assume that those packages are not longer maintained.
So as you showed me, it seems not a problem of nuxt instead its a lack of the plugins regarding mermaid.

... I close this issue as it is no longer a topic regarding the nuxt docs

**Comment by derHodrig** [comment]
I got it to work.
Firstly we need MermaidJs
npm i mermaid
or
yarn add mermaid


Secondly create a new Component in components>global

_mermaid.vue_
```vue
<template>
  <div class="mermaid">
    <slot></slot>
  </div>
</template>

<script>
export default {
  mounted() {
    import("mermaid/dist/mermaid").then(m => {
      m.initialize({
        startOnLoad: true
      });
      m.init();
    });
  }
};
</script>
```

After that you can use it in your markdown

```markdown
<mermaid>
graph TD
  A[Silvester] -->|Get money| B(Go shopping)
  B --> C{Let me think}
  C -->|One| D[Laptop]
  C -->|Two| E[iPhone]
  C -->|Three| F[Car]
  C -->|Four| F[Mac]
</mermaid>
```

**hint: between those <mermaid> tags should no empty lines!**

In the end, it look like this

![own mermaid](https://user-images.githubusercontent.com/53934248/107360199-e5264180-6ad5-11eb-97e5-202c468a39f1.png)

---

No plugin or nuxt.config.js configuration needed

**Comment by ManasMadrecha** [comment]
@derHodrig Your `<mermaid>` component works perfectly. Thanks for this. 💛

But instead of this component, I thought of using the already available `remark-mermaid`, so I created a plugin to register the normal `mermaid.js` inside `/plugins` folder.

```js
let mermaid = require('mermaid/dist/mermaid');

mermaid.initialize({startOnLoad: true })

if (process.client) {
  window.onNuxtReady(() => {
    mermaid.init()
  })
}
```

I registered this plugin inside `nuxtconfig.js`.

**I got an issue:** 
When I write any mermaid syntax inside the code block in markdown, it shows diagram only on hard reload. If I navigate to the page via nuxt-link, I see no diagram. Then, when I hit refresh again, then only the diagram comes back. Any suggestion?
